OSDOCS-12183: adding how to delete manifests (#83246)
openshift/openshift-docs: modules/microshift-manifests-deletion-overview.adoc

// Module included in the following assemblies:
//
// * microshift//running_applications/microshift-deleting-resource-manifests.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-manifests-deletion-overview_{context}"]

> = How manifest deletion works

Reviewer: I might move this module to right before the _Use cases_ module. It explains a lot, about what these subdirectories are, what Microshift is doing, and so forth.

Author: done

> By default, {microshift-short} expects to find deletion manifests in the `delete` subdirectories within the manifests path. When a manifest is placed in these subdirectories, {microshift-short} removes the manifests when the system is started. Read through the following to understand how manifests deletion works in {microshift-short}.

Author: done

> . At every start, before applying the manifests, {microshift-short} scans the following `delete` sub directories within the configured manifests directory to identify the Kustomize manifests that need to be deleted:

Reviewer: I agree that "start" reads awkward for some reason. Is this right? I think it agrees with the intro paragraph.
```suggestion
. Each time the system starts, before applying the manifests, {microshift-short} scans the following `delete` subdirectories within the configured manifests directory to identify the Kustomize manifests that need to be deleted:
```

Author: done

// Module included in the following assemblies:
//
// * microshift//running_applications/microshift-deleting-resource-manifests.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-manifests-deletion-overview_{context}"]
= How manifest deletion works

By default, {microshift-short} expects to find deletion manifests in the `delete` subdirectories within the manifests path. When a manifest is placed in these subdirectories, {microshift-short} automatically handles the removal at every start. Read through the following procedure to understand how manifests deletion works in {microshift-short}.

.Procedure

. At every start, before applying the manifests, {microshift-short} scans the following `delete` sub directories within the configured manifests directory to identify the Kustomize manifests that need to be deleted:

* /usr/lib/microshift/manifests/delete
* /usr/lib/microshift/manifests.d/delete/*
* /etc/microshift/manifests/delete
* /etc/microshift/manifests.d/delete/*

> . {microshift-short} deletes the resources defined in the kustomize manifests found in the `delete` directories by running the equivalent of `kubectl delete -k` command

Reviewer 2: Do we want to include `ignore-not-found` or mention that repeated attempt to delete a manifest does not result in an error?
```suggestion
. {microshift-short} deletes the resources defined in the kustomize manifests found in the `delete` directories by running the equivalent of `kubectl delete --ignore-not-found -k` command
```

Author: done. note is added too.

// Module included in the following assemblies:
//
// * microshift//running_applications/microshift-deleting-resource-manifests.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-manifests-deletion-overview_{context}"]

> = How manifest deletion works

Author: done

> By default, {microshift-short} expects to find deletion manifests in the `delete` subdirectories within the manifest path. When a manifest file is placed in these subdirectories, {microshift-short} automatically handles the removal. Read through the following procedure to understand how manifest file deletion works in {microshift-short}.

Reviewer 2: It handles deletion automatically, but only during start

```suggestion
By default, {microshift-short} expects to find deletion manifests in the `delete` subdirectories within the manifest path. When a manifest file is placed in these subdirectories, {microshift-short} automatically handles the removal during start. Read through the following procedure to understand how manifest deletion works in {microshift-short}.
```

Author: done

> By default, {microshift-short} expects to find deletion manifests in the `delete` subdirectories within the manifests path. When a manifest is placed in these subdirectories, {microshift-short} automatically handles the removal at every start. Read through the following procedure to understand how manifests deletion works in {microshift-short}.

Author: "at every start" is changed but the last sentence is retained.

> .Procedure

Author: done

> . At every start, before applying the manifests, {microshift-short} scans the following `delete` sub directories within the configured manifests directory to identify the Kustomize manifests that need to be deleted:

Author: yes

> . At every start, before applying the manifests, {microshift-short} scans the following `delete` sub directories within the configured manifests directory to identify the Kustomize manifests that need to be deleted:

Reviewer: I agree that "start" isn't clear. Is this right?
```suggestion
. Each time the {microshift-short} service starts, before applying the manifests, {microshift-short} scans the following `delete` subdirectories within the configured manifests directory to identify the Kustomize manifests that need to be deleted:
```

* /usr/lib/microshift/manifests/delete
* /usr/lib/microshift/manifests.d/delete/*
* /etc/microshift/manifests/delete
* /etc/microshift/manifests.d/delete/*

> . {microshift-short} deletes the resources defined in the kustomize manifests found in the `delete` directories by running the equivalent of `kubectl delete --ignore-not-found -k` command.

Author: Kustomize is used throughout

+
[NOTE]
====

> Repeated attempts to delete a manifest do not result in an error.

Author: done. Removed
